### ISSUE TITLE
Bug: editorState null serializes to empty editor state and can no longer be loaded

### ISSUE THREAD

**Issue by johnnyoshika** [facebook/lexical#5079]
Lexical version: 0.12.2

## Steps To Reproduce

1. Load lexical editor with initial `editorState`: `null`. `null` is valid for initial editorState according to it's type definition: https://github.com/facebook/lexical/blob/02f01f160e8894f133e076ab5e38e7a3fd6a5f7a/packages/lexical-react/src/LexicalComposer.tsx#L35-L39

```
    <LexicalComposer
      initialConfig={{
        editorState: null,
        namespace: 'Test',
        theme: {},
        onError: (error: Error) => {
          throw error;
        },
      }}
    >
        <RichTextPlugin
          contentEditable={<ContentEditable />}
          placeholder={
            <p>Enter some text...</p>
          }
          ErrorBoundary={LexicalErrorBoundary}
        />
    </LexicalComposer>
```

2. Serialize editorState to localStorage

```
    localStorage.setItem(
      'editorState',
      JSON.stringify(editor.getEditorState()),
    );
```

The serialized string will be:

```
{"root":{"children":[],"direction":null,"format":"","indent":0,"type":"root","version":1}}
```

3. Deserialize editorState from localStorage

```
    const editorState = localStorage.getItem('editorState');

    if (editorState)
      editor.setEditorState(editor.parseEditorState(editorState));
```

This will result in this exception:

> Uncaught Error: setEditorState: the editor state is empty. Ensure the editor state's root node never becomes empty.

The current workaround is to make sure the initial editorState is `undefined` and not `null`, as `undefined` will auto-create a child paragraph node when the editor is initialized:

```
    <LexicalComposer
      initialConfig={{
        editorState: initialEditorState ?? undefined,
        namespace: 'Test',
        theme: {},
        onError: (error: Error) => {
          throw error;
        },
      }}
    >
```

Link to code example:

https://codesandbox.io/s/lexical-empty-editor-state-ynljf9

After loading the codesandbox, don't type anything into the editor, then click `Save`, then click `Load`. You will see the error appear.

## The current behavior

Editor should load even with empty editorState or passing in null to editorState should not create an empty editorState.

## The expected behavior

No exception to occur even if null is passed in to editorState OR don't allow `null` to be set for initial editorState: https://github.com/facebook/lexical/blob/02f01f160e8894f133e076ab5e38e7a3fd6a5f7a/packages/lexical-react/src/LexicalComposer.tsx#L35-L39

**Comment by acywatson** [comment]
Actually, checking with the team, this is intentional. These values have different semantics - null means to not initialize the editor at all, whereas an undefined value gives the default. The former is used for collaborative editing scenarios.

**Comment by johnnyoshika** [comment]
> > ```
> > .stringify(editor.getEditorState()),
> > ```
> 
> 
> This is not the correct way to serialize. Please read this: https://lexical.dev/docs/concepts/serialization#lexical---json
> 
> 
> Note that call to EditorState.toJSON() before stringifying.

@acywatson  My sample is directly from your documentation that you linked to here: https://lexical.dev/docs/concepts/serialization#lexical---json

![image](https://github.com/facebook/lexical/assets/504505/1cbe14ce-09e1-4e77-bffe-e3603c8efdf6)

Is the documentation incorrect?

**Comment by acywatson** [comment]
No, you're actually right - I forgot that the implementation of JSON.stringify will stringify the result of the toJSON method on the passed in object if that method exists, so these are functionally equivalent.

**Comment by acywatson** [comment]
I'm gonna re-open this as we should update the docs here
